gateway/sds_gateway/users/views.py

```diff
@@ -1800,3 +1819,562 @@ def get(self, request, *args, **kwargs) -> JsonResponse:
+
+    def post(self, request, *args, **kwargs):
+        (
+            files,
+            relative_paths,
+            all_relative_paths,
+            channels,
+            scan_group,
+            capture_type,
+        ) = self._parse_upload_request(request)
+
+        saved_files, errors = self._handle_file_uploads(request, files, relative_paths)
+
+        created_captures = []
+        file_uuids = [f.get("uuid") for f in saved_files if f.get("uuid")]
+
+        # Check if all files were empty (skipped)
+        # If no files were sent (all skipped on frontend), consider them all empty
+        all_files_empty = all(f.size == 0 for f in files) if files else True
+
+        # Create captures if:
+        # 1. All uploads succeeded, OR
+        # 2. We have required fields (regardless of file upload status)
+        capture_errors = []
+        # Check if we have the required fields for capture creation
+        has_required_fields = self._check_required_fields(
+            capture_type, channels, scan_group
+        )
+
+        # Check if this is a chunked upload (skip capture creation for chunks)
+        is_chunk = request.POST.get("is_chunk", "false").lower() == "true"
+        chunk_number = request.POST.get("chunk_number", None)
+        total_chunks = request.POST.get("total_chunks", None)
+
+        # Only create captures if this is not a chunk OR if this is the final chunk
+        should_create_captures = not is_chunk or (
+            chunk_number and total_chunks and int(chunk_number) == int(total_chunks)
+        )
+
+        created_captures = []
+        capture_errors = []
+
+        if should_create_captures:
+            # Handle capture creation
+            created_captures, capture_errors = self._handle_capture_creation(
+                request,
+                channels,
+                relative_paths,
+                capture_type,
+                scan_group,
+                saved_files,
+                all_files_empty,
+                has_required_fields,
+                all_relative_paths,
+            )
+        else:
+            logger.info(
+                "Skipping capture creation for chunk %s of %s",
+                chunk_number,
+                total_chunks,
+            )
+
+        # Link all uploaded files to all created captures
+        # (only if files were actually uploaded)
+        if saved_files and created_captures:
+            # Get File objects by uuid
+            file_objs = File.objects.filter(uuid__in=file_uuids)
+            capture_uuids = [c["uuid"] for c in created_captures if c.get("uuid")]
+            captures = Capture.objects.filter(uuid__in=capture_uuids)
+            for capture in captures:
+                file_objs.update(capture=capture)
+        elif errors and not all_files_empty:
+            logger.error(
+                "File upload errors occurred, skipping capture creation. Errors: %s",
+                errors,
+            )
```

Comment:
Files are already linked to captures during ingestion, we're not doing that manually, nor allowing arbitrary files to be linked - that'd be the role of a dataset.

We can probably drop this selected block and adjust the lines below.